### PR TITLE
PSI functionality into the jupyter notebook example

### LINKED ISSUE #2: PSI
## Description
Apply Private Set Intersection (PSI) to link vertically split data from multiple sources

## Why?
To complete this Epic, we need to generate datasets which simulate a real-world setting: datapoints not shared by both datasets, data appearing in random orders, IDs which link data. We then apply PSI to re-link the data

## Breakdown

- [x] Jumble data #7 
- [x] Add unique IDs to data #8
- [x] Randomly missing data #9
- [x] Add PSI code #13 
- [x] Apply PSI to re-link data #34 

## Who else?
May require changes to https://github.com/OpenMined/PSI

### LINKED ISSUE #34: Apply PSI to SplitNN training example
## Feature Description
Apply the PSI code to the SplitNN notebook to re-link vertically partitioned data. In its current state, the partitioned data is ordered, so PSI is unnecessary; change the arguments passed to the partition function to shuffle the partitioned data.

- Change the parameters passed to the partition function so partitioned data is unordered
- Demonstrate in the notebook that data is unordered i.e. print part of the data (this is not functional, but it helps to tell the story of what we're trying to achieve)
- Use the PSI code to re-order the data
- Train the SplitNN

## Additional Context
Incorporating PSI into the SplitNN training example will finish the PoC
Do not work on this issue until #32 refactor is complete

### PR DESCRIPTION
**Pull request by pavlos-p**
## Description
Added PSI functionality into the the Jupyter Notebook example. 

- The partitioned data now is unordered 
- PSI re-orders the data
- Demonstration in the notebook that firstly data is unordered and PSI orders it (plotted the first 10 images and their labels).

Closes #2 
Closes #34 

## The Test Script Fail
There is currently a bug in the test script during the `bazel build` which causes the tests to fail. 

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: examples/Simple Vertically Partitioned SplitNN.ipynb [hunk 1]
```diff
@@ -135,26 +127,34 @@
...
     "# Batch data\n",
-    "dataloader = VerticalDataLoader(data, batch_size=128) # partition_dataset uses by default \"remove_data=False, keep_order=True\""
+    "dataloader = VerticalDataLoader(data, batch_size=128) # partition_dataset uses by default \"remove_data=True, keep_order=False\"\n",
+    "\n",
+    "# Compute private set intersections\n",
+    "intersection1 = compute_psi(dataloader.dataloader1.dataset.get_ids(), dataloader.dataloader2.dataset.get_ids())\n",
+    "intersection2 = compute_psi(dataloader.dataloader2.dataset.get_ids(), dataloader.dataloader1.dataset.get_ids())\n",
+    "\n",
+    "# Order data\n",
+    "dataloader.drop_non_intersecting(intersection1, intersection2)"
```
Comment (TTitcombe) [inline examples/Simple Vertically Partitioned SplitNN.ipynb:152]: Could you print out the first ~10 entries of each dataset (plotting the images) before and after `drop_non_intersecting`, so we can see that the datasets have been aligned

**Comment by TTitcombe** [comment]
Looks great.

Works locally. Ignore failing builds due to known error
